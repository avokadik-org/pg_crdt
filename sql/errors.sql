set client_min_messages = 'WARNING'; -- pragma:hide
create extension if not exists automerge;
set search_path to public,automerge;

select '{'::autodoc;
select '}'::autodoc;
select '[]'::autodoc;
select '42'::autodoc;
select '"foo"'::autodoc;
select get_int('{"foo": [1]}', 'foo');
select get_int('{"foo": [1]}', '.foo[1]');
select get_int('{"foo": {"bar":[1]}}', '.foo.bar');
select get_int('{"foo": {"bar":[1]}}', '.foo.bar.');
select get_int('{"foo": {"bar":[1]}}', '.foo.bar.[0]');
select get_int('{"foo": {"bar":[1]}}', '.foo.bar[baz]');
