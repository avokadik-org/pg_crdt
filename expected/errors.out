set client_min_messages = 'WARNING'; -- pragma:hide
create extension if not exists automerge;
set search_path to public,automerge;
select '{'::autodoc;
ERROR:  invalid input syntax for type json
LINE 1: select '{'::autodoc;
               ^
DETAIL:  The input string ended unexpectedly.
CONTEXT:  JSON data, line 1: {
select '}'::autodoc;
ERROR:  Error; unable to parse chunk: not enough data.

LINE 1: select '}'::autodoc;
               ^
select '[]'::autodoc;
ERROR:  Error; unable to parse chunk: not enough data.

LINE 1: select '[]'::autodoc;
               ^
select '42'::autodoc;
ERROR:  Error; unable to parse chunk: not enough data.

LINE 1: select '42'::autodoc;
               ^
select '"foo"'::autodoc;
ERROR:  Error; unable to parse chunk: failed to parse header: Invalid magic bytes.

LINE 1: select '"foo"'::autodoc;
               ^
select get_int('{"foo": [1]}', 'foo');
ERROR:  Root path must start with .
select get_int('{"foo": [1]}', '.foo[1]');
ERROR:  Error; Invalid pos 1.

select get_int('{"foo": {"bar":[1]}}', '.foo.bar');
ERROR:  No value found for path: 

select get_int('{"foo": {"bar":[1]}}', '.foo.bar.');
ERROR:  Container for key token . must be map.

select get_int('{"foo": {"bar":[1]}}', '.foo.bar.[0]');
ERROR:  Container for key token .[0] must be map.

select get_int('{"foo": {"bar":[1]}}', '.foo.bar[baz]');
ERROR:  Error: Invalid array index 'baz'

